src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java

```diff
@@ -260,11 +308,23 @@ else if (exercise instanceof TextExercise) {
             submission = new FileUploadSubmission();
         }
 
+        if (submissionType != null) {
```

Comment:
You can remove the if-check. This method creates a new submission with `type=null` anyway.

Reply:
You are right. I removed it.